server/memcached/src/main/java/org/infinispan/server/memcached/MemcachedBaseDecoder.java

```diff
@@ -106,12 +98,44 @@ protected final void exceptionCaught(Header header, Throwable t) {
       }
    }
 
+   private void initializeHandler() {
+      AdvancedCache<byte[], byte[]> c = createCache(server);
+      this.cache = c.withSubject(subject);
+      ComponentRegistry registry = ComponentRegistry.of(cache);
+      VersionGenerator versionGenerator = registry.getComponent(VersionGenerator.class);
+      if (versionGenerator == null) {
+         versionGenerator = new NumericVersionGenerator();
+         registry.registerComponent(versionGenerator, VersionGenerator.class);
+      }
+      this.versionGenerator = versionGenerator;
+      this.timeService = registry.getTimeService();
+      this.statistics = server.getStatistics();
+      this.statsEnabled = statistics != null;
+   }
+
+   protected abstract AdvancedCache<byte[], byte[]> createCache(MemcachedServer server);
+
    @Override
    public void handlerAdded(ChannelHandlerContext ctx) throws Exception {
       this.ctx = ctx;
       super.handlerAdded(ctx);
    }
 
+   @Override
+   public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exception {
```

Comment:
I feel like this initialization code should sit in an abstract class that all the other decoders implement. Otherwise, we may miss an update in one of the classes. Can you see if that is possible to consolidate this code?

Reply:
It doesn't seem likely. `MemcachedBaseDecoder` is extending `ByteToMessageDecoder`, and `RespHandler` is extending `ChannelInboundHandlerAdapter`. In the Memcached impl, this class is the one for backpressure and decoding. In the Resp impl, we have a handler to decode, and this handler for backpressure.

Reply:
Sounds good, let me see. I'll update to handle the comment above and will check this.

Reply:
I guess the current is okay. I was just hoping for a way to consolidate the code to prevent possible further issues, but sometimes that doesn't work :(